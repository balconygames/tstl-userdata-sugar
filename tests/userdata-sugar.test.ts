import * as tstl from "typescript-to-lua";
import path from "path";
import { readFileSync } from "fs";

it("spreads userdata array", async () => {
	const result = tstl.transpileProject(path.join(__dirname, "tests.tsconfig.json"));
	expect(result.diagnostics).toHaveLength(0);

	const lua = readFileSync(path.join(__dirname, "out", "tests", "test.lua"), "utf-8");
    expect(lua).toMatchSnapshot();
});
