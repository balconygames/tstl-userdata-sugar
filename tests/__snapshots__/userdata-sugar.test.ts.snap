// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`spreads userdata array 1`] = `
"--[[ Generated with https://github.com/TypeScriptToLua/TypeScriptToLua ]]
-- Lua Library inline imports
function __TS__ArrayMap(arr, callbackfn)
    local newArray = {}
    do
        local i = 0
        while i < #arr do
            newArray[i + 1] = callbackfn(_G, arr[i + 1], i, arr)
            i = i + 1
        end
    end
    return newArray
end

userData = makeUserData()
print(
    {
        unpack(
            __TS__ArrayMap(
                userData,
                function(___, x) return x end
            )
        )
    }
)
"
`;
